### PR TITLE
Include request_id in HTTP request logs

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This adds the request_id to the HTTP request logs. I also altered the log message to be more consistent to other log messages.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-19550

### CODE REVIEW

File: web/handlers.go [hunk 1]
```diff
@@ -72,14 +72,16 @@ type Handler struct {
 
 func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	now := time.Now()
-	mlog.Debug("request:", mlog.String("method", r.Method), mlog.String("url", r.URL.Path))
+
+	requestID := model.NewId()
+	mlog.Debug("New HTTP request", mlog.String("method", r.Method), mlog.String("url", r.URL.Path), mlog.String("request_id", requestID))
```
Comment (lieut-data) [inline web/handlers.go]: The word `New` here doesn't add much value for me. Perhaps:
```suggestion
	mlog.Debug("HTTP request", mlog.String("method", r.Method), mlog.String("url", r.URL.Path), mlog.String("request_id", requestID))
```

Comment (hanzei) [reply web/handlers.go]: What do you think about
```suggestion
	mlog.Debug("Received HTTP request", mlog.String("method", r.Method), mlog.String("url", r.URL.Path), mlog.String("request_id", requestID))
```

**Comment by hanzei** [comment]
> . Also, this might be a stretch, but is there any way to unit test this?

I don't know any tests that asserts a specific log message. Do you know one?

This is a bit tricky to implement, because we use one global logger (`mlog.globalLogger`) and I don't see a straight forward way to change this.

Also, asserting log statements is a bit weird, because the code is often straight  forward. Like in this case.

**Comment by lieut-data** [comment]
Happy to skip the unit tests for a case like this as you suggest. Does it make sense to update the text as per the suggestion above?

File: web/handlers.go (hunk 1 above)
Comment (hanzei) [reply web/handlers.go]: What do you think about
```suggestion
	mlog.Debug("Received HTTP request", mlog.String("method", r.Method), mlog.String("url", r.URL.Path), mlog.String("request_id", requestID))
```

**Comment by hanzei** [comment]
I swear I leaved a response to your suggestion... Please take a look

**Comment by lieut-data** [comment]
Works for me :)

**Comment by lieut-data** [comment]
@hanzei, as small as this is, this would be something good to demo at the next R&D, since it might solve a painpoint folks haven't even realized they had had yet :)

**Pull request merged**